symfony/src/Controller/AccessManagementController.php

```diff
@@ -32,16 +47,102 @@ public function userGroups()
+            if ($groupMemberForm->isValid()) {
+                $email = $groupMemberForm->get('email')->getData() . self::MEMBER_DOMAIN;
+                $result = $this->googleGroupsService->addMember($group->email, $email);
+                if ($result['status'] === 'success') {
+                    $this->addFlash('success', $result['message']);
+                    $this->loggerService->log(Log::GROUP_MEMBER_ADD, [
+                        'member' => $email,
+                        'result' => 'success'
+                    ]);
+                    return $this->redirectToRoute('access_manage_user_group', ['groupId' => $groupId]);
+                }
+                $this->addFlash('error', $result['message']);
+                $this->loggerService->log(Log::GROUP_MEMBER_ADD, [
+                    'member' => $email,
+                    'result' => 'fail'
+                ]);
+            } else {
+                $groupMemberForm->addError(new FormError('Please correct the errors below.'));
+            }
+        }
+        return $this->render('accessmanagement/add-member.html.twig', [
+            'group' => $group,
+            'groupMemberForm' => $groupMemberForm->createView()
+        ]);
+    }
+
+    /**
+     * @Route("/user/group/{groupId}/member/{memberId}/remove", name="access_manage_user_group_remove_member")
+     */
+    public function removeMember($groupId, $memberId, Request $request)
+    {
+        $group = $this->getUser()->getSiteFromId($groupId);
+        if (empty($group)) {
+            throw $this->createNotFoundException();
+        }
+        $member = $this->googleGroupsService->getMemberById($group->email, $memberId);
+        if (empty($member)) {
+            throw $this->createNotFoundException();
+        }
+        $removeGoupMemberForm = $this->createForm(RemoveGroupMemberType::class);
+        $removeGoupMemberForm->handleRequest($request);
+        if ($removeGoupMemberForm->isSubmitted()) {
+            if ($removeGoupMemberForm->isValid()) {
+                $confirm = $removeGoupMemberForm->get('confirm')->getData();
+                if ($confirm === 'yes') {
+                    $result = $this->googleGroupsService->removeMember($group->email, $member->email);
+                    if ($result['status'] === 'success') {
+                        $this->addFlash('success', $result['message']);
+                        $this->loggerService->log(Log::GROUP_MEMBER_REMOVE, [
+                            'member' => $member->email,
+                            'result' => 'success'
+                        ]);
+                        return $this->redirectToRoute('access_manage_user_group', ['groupId' => $groupId]);
+                    }
+                    $this->addFlash('error', $result['message']);
+                    $this->loggerService->log(Log::GROUP_MEMBER_REMOVE, [
+                        'member' => $member->email,
+                        'result' => 'fail'
+                    ]);
```

Comment:
I'm not sure we want to return the error message to the user. In the `removeMember` method, you are catching an API exception and returning the message. That may contain information that we don't want to display and/or could be confusing. I think a generic messsage could be added to the flash message, and then we can add the result message to the log object.